src/classes/frDonation.cls

```diff
@@ -62,76 +62,180 @@ public class frDonation extends frModel implements frSyncable {
     public Boolean sync() {
         Boolean result = false;
         Map<String, Object> request = getRequestBody();
-        o = new Opportunity();
-        applyMappings(o, request);
-        setOpportunityStage(o, String.valueOf(request.get('status')));
-        
-        String funraiseId = getFunraiseId();       
-        String supporterFunraiseId = String.valueOf(request.get('donorId'));        
-        if (String.isNotBlank(supporterFunraiseId)) {
-            List<Contact> contacts = [SELECT Id, AccountId FROM Contact WHERE fr_ID__c = :supporterFunraiseId];
-            if (contacts.size() > 0) {
-                Contact supporterContact = contacts.get(0);
-                o.fr_Donor__c = supporterContact.Id;
-                if(frUtil.hasNPSP()){
-                    o.put('npsp__Primary_Contact__c', supporterContact.Id);
-                    o.AccountId = supporterContact.AccountId;
+        String frId = getFunraiseId();
+        SObject gt = null;
+        for (String key : request.keySet()) {
+            Object value = request.get(key);
+            System.debug('Key: ' + key + ', Value: ' + value);
+        }
+        // Gift Transaction Split
+        if (Schema.getGlobalDescribe().containsKey('gifttransaction')) {
+            //add a check for a dupe transaction?
+            String objectName = 'gifttransaction';  
+            List<SObject> gtList = Database.query('SELECT Id, Name FROM GiftTransaction WHERE fr_ID__c = :frId');
```

Comment:
How will we be able to create the fr_ID__C field on the GiftTransaction object?  

Can we include that custom field in this repository & will that still be valid to deploy to organizations that do not have a GiftTransaction object?

Reply:
The idea was not put any fields or objects relating to NPC objects as in the code otherwise it'll be a package dependency.  That field would need to be created and mapped post install.  We added validation on the mapping screens that force a field with the API fr_ID__c to exist and to be mapped to allow a successful save.  There are probably some ways to make the setup more elegant but that was the intent.